File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/1678. Goal Parser Interpretation/Python solution/1678. Goal Parser Interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        out = ""
        i = 0
        while i < len(command):
            if command[i] == 'G':
                out += "G"
            elif command[i] == '(' and command[i + 1] == ')':
                out += "o"
            elif command[i] == '(' and command[i + 1] == 'a':
                out += "al"
            i += 1
        return out

        # Or simply use the replace method 
        # return command.replace("()", "o").replace("(al)", "al")
        

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/1678. Goal Parser Interpretation/C++ solution/1678. Goal Parser Interpretation.cpp
class Solution {
public:
    string interpret(string command) {
        string out = "";
        int i = 0;
        while(i < command.length())
        {
            if (command[i] == 'G')
                out += "G";
            else if (command[i] == '(' and command[i + 1] == ')')
                out += "o";
            else if (command[i] == '(' and command[i + 1] == 'a')
                out += "al";
            i++;
        }
        return out;
    }
};